src/app/src/main/kotlin/org/imperial/mrc/hint/security/oauth2/OAuth2StateGenerator.kt

```diff
@@ -0,0 +1,23 @@
+package org.imperial.mrc.hint.security.oauth2
+
+import java.util.*
+
+object OAuth2StateGenerator
+{
+    private val randomString = UUID.randomUUID().toString()
+
+    fun plaintextState(): String
+    {
+        return randomString
+    }
+
+    fun encodedState(): String
+    {
+        return Base64.getEncoder().encodeToString(randomString.toByteArray())
+    }
+
+    fun decodedState(code: String): String
+    {
+        return Base64.getDecoder().decode(code).decodeToString()
+    }
```

Comment:
Why is the base64 encoding needed? Is it because otherwise the hyphens would get url encoded in the query string so the comparison would fail? A comment here would be appreciated too! :)